js/turtleactions/IntervalsActions.js

```diff
@@ -72,36 +73,70 @@ function setupIntervalsActions(activity) {
          */
         static GetIntervalNumber(turtle) {
             const tur = activity.turtles.ithTurtle(turtle);
-            const noteStatus = tur.singer.noteStatus;
-            const notePitches = tur.singer.notePitches;
-            let a='C', b='C', octave=0;
-            
-            
-            if (noteStatus && noteStatus[0])
-            {
-                if (!noteStatus[0][1]) return 0;   
-                    a = noteStatus[0][0].substring(0, noteStatus[0][0].length - 1);
-            b = noteStatus[0][1].substring(0, noteStatus[0][1].length - 1);
-            const octavea = parseInt(noteStatus[0][0].substring(noteStatus[0][0].length - 1));
-            const octaveb = parseInt(noteStatus[0][1].substring(noteStatus[0][1].length - 1));
-            octave = octaveb - octavea;
-            } else if (notePitches) {
-                const pitchBlk = notePitches[last(tur.singer.inNoteBlock)];
-                a = pitchBlk[0];
-                b = pitchBlk[pitchBlk.length - 1];
-            }
-            
-            
-            let totalIntervals = Math.abs(NOTESTEP[a]-NOTESTEP[b]);
-            
-            while (octave) {
+            let { firstNote, secondNote, octave } = GetNotesForInterval(tur);
+            let totalIntervals = Math.abs(ALLNOTESTEP[firstNote] - ALLNOTESTEP[secondNote]);
+            if (ALLNOTESTEP[secondNote] < ALLNOTESTEP[firstNote]) totalIntervals = 12 - totalIntervals;
+            if (octave<0&&totalIntervals!==0&&totalIntervals!==12)  totalIntervals = 12 - totalIntervals;
+            if(octave<-1||totalIntervals===0)octave = Math.abs(octave);
+            while (octave > 0) {
                 totalIntervals += 12;
                 octave--;
             }
-            
             return totalIntervals;
         }
 
+         /**
+         * @static
+         * @param {number} turtle
+         * @returns {String}
+         */
+
+        static GetCurrentInterval(turtle) {
+            const tur = activity.turtles.ithTurtle(turtle);
+
+            const { firstNote, secondNote, octave } = GetNotesForInterval(tur);
+
+            const index1 = NOTENAMES.indexOf(firstNote.substring(0, 1));
+            const index2 = NOTENAMES.indexOf(secondNote.substring(0, 1));
+            let lastWord = "";
+            let letterGap = Math.abs(index2 - index1);
+            if (index1 > index2) letterGap = NOTENAMES.length - letterGap;
+
+            let totalIntervals = this.GetIntervalNumber(turtle);
+            const numberToStringMap = ['an', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
```

Comment:
"an" is not going to work well with translation. Please change to "one".

Note that any string that is to be translated needs to be inside _(), e.g., _("two")

Reply:
@walterbender should i just do this _() to  the return value ?

Reply:
Better to add it to each string. Otherwise they won't be added to the list of strings to translate

Reply:
I have done  the changes